src/RMGVertexConfinement.cc

```diff
@@ -121,8 +121,125 @@ bool RMGVertexConfinement::SampleableObject::IsInside(const G4ThreeVector& verte
 
   return false;
 }
+
+
+// get the intersections between the solid and an arbitrary line, defined by start and dir
+std::vector<G4ThreeVector> RMGVertexConfinement::SampleableObject::GetIntersections(G4ThreeVector start,
+    G4ThreeVector dir) const {
+
+
+  // check if the physical volume exists
+  if ((not this->physical_volume) or (not this->physical_volume->GetLogicalVolume()->GetSolid()))
+    RMGLog::OutDev(RMGLog::fatal, "Cannot find number of intersections for a SampleableObject where the physical volume is not set",
+        "this probably means you are trying to generically sample a geometrical volume which "
+        "instead",
+        "should be natively sampled");
+
+  auto solid = this->physical_volume->GetLogicalVolume()->GetSolid();
+
+
+  G4double dist = 0;
+  std::vector<G4ThreeVector> intersections = {};
+  G4ThreeVector int_point;
+
+  int_point = start;
+  int counter = 0;
+
+  dist = 0;
+
+  while (dist < kInfinity) {
+
+    dist = (counter % 2) == 0 ? solid->DistanceToIn(int_point, dir)
+                              : solid->DistanceToOut(int_point, dir);
+
+    int_point = int_point + dist * dir;
+
+    if (dist < kInfinity) intersections.push_back(int_point);
+    counter++;
+  }
+
+  if (intersections.size() % 2) throw;
+
+  return intersections;
+}
+
+
+// generate a random position and direction for the surface sampling
+void RMGVertexConfinement::SampleableObject::GetDirection(G4ThreeVector& dir,
+    G4ThreeVector& pos) const {
+
+
+  if ((not this->physical_volume) or (not this->physical_volume->GetLogicalVolume()->GetSolid()))
+    RMGLog::OutDev(RMGLog::fatal, "Cannot find number of intersections for a SampleableObject where the physical volume is not set",
+        "this probably means you are trying to generically sample a geometrical volume which "
+        "instead",
+        "should be natively sampled");
+
+  // get the bounding radius
+  G4double bounding_radius =
+      physical_volume->GetLogicalVolume()->GetSolid()->GetExtent().GetExtentRadius();
+
+
+  // start on z-axis, pointing down.
+  pos = G4ThreeVector(0.0, 0.0, bounding_radius);
+  dir = G4ThreeVector(0.0, 0.0, -1.0);
+
+  // push in rho direction by some impact parameter
+  G4double diskPhi = 2.0 * CLHEP::pi * G4UniformRand();
+  G4double diskR = sqrt(G4UniformRand()) * bounding_radius;
```

Comment:
```suggestion
  G4double disk_phi = 2.0 * CLHEP::pi * G4UniformRand();
  G4double disk_r = sqrt(G4UniformRand()) * bounding_radius;
```
we should use snake casing, except for class members where we use `fMember`.